book/src/getting_started.md

```diff
@@ -0,0 +1,27 @@
+# Want to run the code?
+
+One of the goals with this repo is to experiment and compare different
+implementations of the same algorithm. You can run the analysis by using `cargo run`
+and you can choose the analysis with `-a`. So for example to run against an example
+extract from clap, you might do:
+
+```bash
+> cargo +nightly run --release -- -a DatafrogOpt inputs/clap-rs/app-parser-{{impl}}-add_defaults/
+    Finished release [optimized] target(s) in 0.05 secs
+     Running `target/release/borrow-check 'inputs/clap-rs/app-parser-{{impl}}-add_defaults/'`
+--------------------------------------------------
+Directory: inputs/clap-rs/app-parser-{{impl}}-add_defaults/
+Time: 3.856s
+```
+
+You could also try `-a Naive` to get the naive rules (more readable,
+slower) -- these are the exact rules described in [the
+blogpost][post]. You can also use `-a LocationInsensitive` to use a
+location insensitive analysis (faster, but may yield spurious errors).
```

Comment:
I wonder if trying `LocationInsensitive` and then `Naive` if the former fails would be faster overal.

Reply:
that already exists, indeed. I think it's called `-a Hybrid`

Reply:
Oh by the way, the [`Hybrid` variant](https://github.com/rust-lang/polonius/blob/master/polonius-engine/src/output/hybrid.rs) calls the optimized variant (`DatafrogOpt`) if it finds possible errors, not the `Naive` one.

This `Hybrid` variant is also the default one that `rustc` uses when calling into Polonius.

We have in-progress work which shares the results of the location insensitive pre-pass with the optimized variant, so that there are only a small number of possible errors to check (and that is a lot faster as well).